src/Diagnostics.DataProviders/DataProviderConfigurations/GenericMdmDataProviderConfiguration.cs

```diff
@@ -32,4 +35,64 @@ public void PostInitialize()
         {
         }
     }
+
+    public class GenericMdmDataProviderConfigurationWrapper: IDataProviderConfiguration, IMdmDataProviderConfiguration
+    {
+        /// <summary>
+        /// Gets or sets the base endpoint.
+        /// </summary>
+        public string Endpoint { get; set; }
+
+        /// <summary>
+        /// Gets or sets the certificate thumbprint.
+        /// </summary>
+        public string CertificateThumbprint { get; set; }
+
+        /// <summary>
+        /// Gets or sets monitoring account.
+        /// </summary>
+        public string MonitoringAccount { get; set; }
+
+        /// <summary>
+        /// Post initialize.
+        /// </summary>
+        public void PostInitialize()
+        {
+        }
+
+        public GenericMdmDataProviderConfigurationWrapper(GenericMdmDataProviderConfiguration config)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException("Configuration is null.");
+            }
+            if (config.MonitoringAccount != null && config.MonitoringAccount.StartsWith("Mock"))
+            {
+                CertificateThumbprint = "BAD0BAD0BAD0BAD0BAD0BAD0BAD0BAD0BAD0BAD0";
+            }
+            else
+            {
+                using (X509Store store = new X509Store(StoreName.My, StoreLocation.LocalMachine))
+                using (X509Store userStore = new X509Store(StoreName.My, StoreLocation.CurrentUser))

```

Comment:
In the master branch, the cert is expected to be present only in the Computer store. No need to iterate through the user store. Iterating certificate store is an expensive operation..